### PR TITLE
Handle KeyboardInterrupt cleanly

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: embuilder.py [hunk 1]
```diff
@@ -278,7 +281,15 @@ def build_port(port_name, lib_name, params):
...
 
-else:
-  shared.logging.error('unfamiliar operation: ' + operation)
-  sys.exit(1)
+
+if __name__ == '__main__':
+  try:
+    sys.exit(main())
+  except KeyboardInterrupt:
+    logging.warngin("KeyboardInterrupt")
```
Comment (jgravelle-google) [inline embuilder.py]: warngin -> warning?

File: tests/runner.py [hunk 2]
```diff
@@ -37,6 +37,8 @@ def path_from_root(*pathelems):
 from tools.shared import *
 from tools.line_endings import check_line_endings
 
+logger = logging.getLogger('tests/runner.py')
```
Comment (jgravelle-google) [inline tests/runner.py]: Why?

Can multiple subprocess reuse the same logger? A comment would help I think

File: embuilder.py [hunk 3]
```diff
@@ -277,19 +279,19 @@ def main():
     elif what == 'cocos2d':
       build_port('cocos2d', None, ['-s', 'USE_COCOS2D=3', '-s', 'USE_ZLIB=1', '-s', 'USE_LIBPNG=1'])
     else:
-      shared.logging.error('unfamiliar build target: ' + what)
```
Comment (jgravelle-google) [inline embuilder.py]: This file still has references to shared.logging, should we want to update those too?

Comment (sbc100) [reply embuilder.py]: Yes. Done.

File: tests/runner.py (hunk 2 above)
Comment (jgravelle-google) [reply tests/runner.py]: I'm still not totally clear on why logger vs logging

Comment (sbc100) [reply tests/runner.py]: logging.xxxx is calling methods on the root logger, which is always called "root".. which means all log messages have the word root in them.   When you have larger programs or programs with which invoke each other (as in this case) is better to have distinct loggers.

The existing code which imported the logging module from the shared module is just plane weird.

Comment (jgravelle-google) [reply tests/runner.py]: Ah, makes sense
Yeah that was weird

**Comment by kripken** [comment]
Fix for an error reporting regression from here is in #6732.

**Comment by sbc100** [comment]
Sorry about that..  How this this change cause the regression?

**Comment by kripken** [comment]
I believe what happened is this PR correctly replaced a `sys.exit` with a `return` in `run_tests()`, however, `main()` which calls `run_tests()` was missing a `return` of that value, so error codes didn't go through.

**Pull request merged**